include/llbuild/BuildSystem/CommandResult.h

```diff
@@ -24,6 +26,32 @@ enum class CommandResult {
   Skipped,
 };
 
+/// Extended result of a command execution.
+struct CommandExtendedResult {
+  CommandResult result; /// The final status of the command
+  int exitStatus;       /// The exit code
+
+  uint64_t utime;       /// User time (in us)
+  uint64_t stime;       /// Sys time (in us)
+  uint64_t maxrss;      /// Max RSS (in bytes)
+
+
+  CommandExtendedResult(CommandResult result, int exitStatus, uint64_t utime = 0,
+                uint64_t stime = 0, uint64_t maxrss = 0)
+    : result(result), exitStatus(exitStatus)
+    , utime(utime), stime(stime), maxrss(maxrss)
+  {}
+
+  static CommandExtendedResult failed(int exitStatus = -1) {
```

Comment:
We have a "mini convention" in LLVM of calling struct factory functions something like `makeFailed` which I kinda prefer, unless you have an objection